service/worker/service.go

```diff
@@ -468,7 +470,12 @@ func (s *Service) startBatcher() {
 	}
 }
 
-func (s *Service) initScanner() {
+func (s *Service) initScanner() error {
+	currentCluster := s.clusterMetadata.GetCurrentClusterName()
+	adminClient, err := s.clientBean.GetRemoteAdminClient(currentCluster)
```

Comment:
I think we should add a GetAdminClient to ClientBean to keep it similar to Get[Remote]FrontendClient, since frontend and admin clients work the same way.

also, I noticed Get/SetFrontendClient in clientBean.go don't do proper locking

Reply:
I plan to include this in the next available patch. So I created an issue to track here: https://github.com/temporalio/temporal/issues/3532.

Reply:
Do you plan to fix locking bug in this PR?

Reply:
No. I don't see caller in those setters. I will remove them in another PR.